sqlmesh/core/snapshot/evaluator.py

```diff
@@ -87,15 +87,25 @@ class SnapshotEvaluator:
     does not directly communicate with the underlying execution engine.
 
     Args:
-        adapter: The adapter that interfaces with the execution engine.
+        adapters: A single EngineAdapter or a dictionary of EngineAdapters where
+            the key is the gateway name of the connection.
         ddl_concurrent_tasks: The number of concurrent tasks used for DDL
             operations (table / view creation, deletion, etc). Default: 1.
     """
 
-    def __init__(self, adapter: EngineAdapter, ddl_concurrent_tasks: int = 1):
-        self.adapter = adapter
+    def __init__(
+        self,
+        adapters: EngineAdapter | t.Dict[str, EngineAdapter],
+        ddl_concurrent_tasks: int = 1,
+    ):
+        self.adapters = adapters if isinstance(adapters, t.Dict) else {"": adapters}
         self.ddl_concurrent_tasks = ddl_concurrent_tasks
 
+    def adapter(self, gateway: t.Optional[str] = None) -> EngineAdapter:
+        """Returns the adapter for the specified gateway or the default adapter if none is provided."""
+        default_gateway = next(iter(self.adapters))
+        return self.adapters.get(gateway or default_gateway, self.adapters[default_gateway])
```

Comment:
should we fail if the gateway was specified but is missing in the mapping?

Reply:
> Is the first gateway in self.adapters always the default gateway? 

yes because in the context, the adapters are set by placing the default gateway as the first item in the dictionary.

> What if the config has a default_gateway, different than the first one?

this could in theory occur if someone initialised a SnapshotEvaluator but didn't set the first item in the dictionary of adapters as the default. so the assumption is that when a dictionary of adapters is passed to the evaluator, the first item is always considered the default gateway. since the default engine adapter is no longer passed explicitly to the evaluator, I was unsure if there might be a better way to handle this

> should we fail if the gateway was specified but is missing in the mapping?

yes, I was unsure whether to fail or not, but I suppose it would be wiser to raise an error rather than default to the adapter